exporter/logzioexporter/exporter.go

```diff
@@ -15,9 +15,97 @@
 package logzioexporter
 
 import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/logzio/jaeger-logzio/store"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.opentelemetry.io/collector/translator/trace/jaeger"
+)
+
+const (
+	loggerName = "logzio-exporter"
 )
 
+// logzioExporter implements an OpenTelemetry trace exporter that exports all spans to Logz.io
+type logzioExporter struct {
+	accountToken string
+	writer       *store.LogzioSpanWriter
+	logger       hclog.Logger
+}
+
+var WriteSpanFunc func(span *model.Span) error
+var InternalTracesToJaegerTraces = jaeger.InternalTracesToJaegerProto
+
+func newLogzioExporter(config *Config, params component.ExporterCreateParams) (*logzioExporter, error) {
+	logger := hclog.New(&hclog.LoggerOptions{
```

Comment:
This should use the zap logger that is passed into the CreateTraceExporter method in the factory under the `params` param.

Reply:
If you need logging capabilities in your logzio client library, make it support an interface or somehow wrap the zap logger.

Reply:
@keitwb Done